src/plugins/embeddable/public/lib/panel/embeddable_panel.tsx

```diff
@@ -113,6 +118,22 @@ export class EmbeddablePanel extends React.Component<Props, State> {
     });
   }
 
+  private async refreshNotifications() {
+    let notifications = await this.props.getActions(PANEL_NOTIFICATION_TRIGGER, {
+      embeddable: this.props.embeddable,
+    });
+    if (!this.mounted) return;
+
+    const { disabledActions } = this.props.embeddable.getInput();
+    if (disabledActions) {
+      notifications = notifications.filter(badge => disabledActions.indexOf(badge.id) === -1);
+    }
```

Comment:
I would like keep this logic in the panel. To me it feels like `.disabledActions ` is the setting of the panel and the panel is one place which makes sure the right actions are rendered. If we move this setting to the actions, we would need to implement this check in every existing action and 3rd party developers would also need to honor this setting. Also, if we move this check to the action it kinda forces the action to know that an embeddable has an input with `. disabledActions` on its input, which would be harder to change in the future if we now make all actions aware of this `.disabledActions` setting.

Tell me if you still want to move it here.

One thing I was thinking about which could allow to do the above suggestion in a nice way is to introduce a formal `Context` class. Something like:

```ts
class Context<Data> {
  trigger: Trigger;
  data: Data;
  disabledActions?: string[];
}
```

And use it when executing triggers.

```ts
plugins.uiActions.getTrigger('...').exec(data);
```

The above wraps `data` into `Context`.

```ts
const context = new Context(data);
```

This way on the `context` object we would always have `.trigger` property, this is useful because actions can be attached to multiple triggers, so that action knows which trigger executed it.

We could also add more functionality to the `Context` class, like `.disabledActions` above, that way the `ActionInternal` could lookup `.disabledActions` automatically and disable the action automatically in all actions, without us needing to implement this check in all actions.

Reply:
Good argument, keeping in here sgtm.  Interesting thoughts on `Context`, I think it could work.  This works too though. 👍